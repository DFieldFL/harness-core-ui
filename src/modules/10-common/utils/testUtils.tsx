import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uikit'
import qs from 'qs'

import strings from 'strings/strings.en.yaml'
import { AppStoreProvider, useAppStoreWriter } from 'framework/hooks/useAppStore'
import type { AppStore } from 'framework/types/AppStore'
import { AUTH_ROUTE_PATH_PREFIX, StringsContext } from 'framework/exports'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): Element | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): Element | null => document.querySelector('.bp3-popover-content')

interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, any>
  defaultAppStoreValues?: Partial<AppStore>
}

export const prependAccountPath = (path: string): string => AUTH_ROUTE_PATH_PREFIX + path

const AppStoreInitializer: React.FC<{ defaultAppStoreValues: TestWrapperProps['defaultAppStoreValues'] }> = ({
  children,
  defaultAppStoreValues
}) => {
  const updateAppStore = useAppStoreWriter()
  updateAppStore(defaultAppStoreValues || {})
  return <> {children} </>
}

export const NotFound = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  const { path = '/', pathParams = {}, defaultAppStoreValues, queryParams = {} } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  const history = createMemoryHistory({ initialEntries: [routePath] })

  return (
    <AppStoreProvider>
      <StringsContext.Provider value={strings}>
        <AppStoreInitializer defaultAppStoreValues={defaultAppStoreValues}>
          <Router history={history}>
            <ModalProvider>
              <RestfulProvider base="/">
                <Switch>
                  <Route exact path={path}>
                    {props.children}
                  </Route>
                  <Route>
                    <NotFound />
                  </Route>
                </Switch>
              </RestfulProvider>
            </ModalProvider>
          </Router>
        </AppStoreInitializer>
      </StringsContext.Provider>
    </AppStoreProvider>
  )
}
