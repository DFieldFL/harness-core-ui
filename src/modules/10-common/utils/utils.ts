import type { IconName } from '@wings-software/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}
