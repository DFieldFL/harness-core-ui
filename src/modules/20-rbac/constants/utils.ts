export enum RbacResourceGroupTypes {
  DYNAMIC_RESOURCE_SELECTOR = 'DynamicResourceSelector'
}
