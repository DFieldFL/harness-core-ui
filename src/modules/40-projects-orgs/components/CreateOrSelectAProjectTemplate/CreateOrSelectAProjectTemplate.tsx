import React from 'react'
import { Text, Layout, Color, Button } from '@wings-software/uicore'
import { useStrings } from 'framework/strings'

interface CreateOrSelectAProjectTemplateProps {
  onSelectProject: () => void
  onCreateProject: () => void
  moduleDescription: string
}

export const CreateOrSelectAProjectTemplate: React.FC<CreateOrSelectAProjectTemplateProps> = ({
  onSelectProject,
  onCreateProject,
  moduleDescription
}) => {
  const { getString } = useStrings()

  return (
    <Layout.Vertical spacing="small">
      <Text style={{ color: Color.BLACK, fontSize: 'large', fontWeight: 'bold' }}>
        {getString('projectsOrgs.createOrSelectAProject')}
      </Text>
      <Text style={{ color: Color.BLACK, fontSize: 'small' }}>
        {getString('projectsOrgs.getStarted', { moduleDescription })}
      </Text>
      <Layout.Vertical spacing="small" padding={{ top: 'large' }}>
        <Button
          intent="primary"
          text={getString('projectsOrgs.selectAnExistingProject')}
          onClick={onSelectProject}
          width="60%"
        />
        <Button
          intent="none"
          text={getString('projectsOrgs.createANewProject')}
          onClick={onCreateProject}
          width="60%"
        />
      </Layout.Vertical>
    </Layout.Vertical>
  )
}
