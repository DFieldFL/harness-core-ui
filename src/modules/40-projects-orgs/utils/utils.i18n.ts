export default {
  descriptionCD: 'Deploy your services with blazingly fast pipelines.',
  descriptionCV: 'Deploy in peace, verify activities that take place in system. Identify risk early.',
  descriptionCE: 'Spot and quickly debug inefficiencies and optimize them to reduce costs.',
  descriptionCI: 'Commit, build, and test your code at a whole new level.',
  descriptionCF: 'Decouple release from deployment and rollout features safely and quickly.'
}
