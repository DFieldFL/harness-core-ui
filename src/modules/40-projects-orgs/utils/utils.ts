import type { IconName } from '@wings-software/uikit'
import { ModuleName } from 'framework/exports'
import type { Project } from 'services/cd-ng'
import i18n from '@projects-orgs/pages/projects/ProjectsPage.i18n'

const allModules: Required<Project>['modules'] = ['CD', 'CV', 'CI', 'CE', 'CF']

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getModulePurpose = (module: ModuleName): string => {
  switch (module) {
    case ModuleName.CD:
      return 'Delivery'
    case ModuleName.CV:
      return 'Verification'
    case ModuleName.CI:
      return 'Integration'
    case ModuleName.CE:
      return 'Efficiency'
    case ModuleName.CF:
      return 'Features'
  }
  return ''
}

export const getModuleDescription = (module: ModuleName): string => {
  switch (module) {
    case ModuleName.CD:
      return i18n.newProjectWizard.purposeList.descriptionCD
    case ModuleName.CV:
      return i18n.newProjectWizard.purposeList.descriptionCV
    case ModuleName.CI:
      return i18n.newProjectWizard.purposeList.descriptionCI
    case ModuleName.CE:
      return i18n.newProjectWizard.purposeList.descriptionCE
    case ModuleName.CF:
      return i18n.newProjectWizard.purposeList.descriptionCF
  }
  return ''
}

export const getEnableModules = (modules: Required<Project>['modules']): Required<Project>['modules'] => {
  return allModules.filter(module => !modules.includes(module))
}
