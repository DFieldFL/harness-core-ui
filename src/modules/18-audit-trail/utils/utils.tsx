import React from 'react'
import { Menu } from '@blueprintjs/core'
import type { ItemRenderer } from '@blueprintjs/select'
import { Layout, Text, Avatar, Color, MultiSelectOption } from '@wings-software/uicore'
import { isEmail } from '@common/utils/Validation'

export interface UserItem extends MultiSelectOption {
  email?: string
}

export const UserItemRenderer: ItemRenderer<UserItem> = (item, { handleClick }) => (
  <Menu.Item
    key={item.value.toString()}
    text={
      <Layout.Horizontal spacing="small" flex={{ alignItems: 'center', justifyContent: 'flex-start' }}>
        <Avatar name={item.label} email={item.email || item.value.toString()} size="small" hoverCard={false} />
        <Layout.Vertical padding={{ left: 'small' }}>
          <Text color={Color.BLACK}>{item.label}</Text>
          <Text color={Color.GREY_700}>{item.email || item.value}</Text>
        </Layout.Vertical>
      </Layout.Horizontal>
    }
    onClick={handleClick}
  />
)

export const UserTagRenderer = (item: UserItem, validate = false): React.ReactNode => (
  <Layout.Horizontal key={item.value.toString()} flex spacing="small">
    <Avatar name={item.label} email={item.value.toString()} size="xsmall" hoverCard={false} />
    <Text color={validate && !isEmail(item.value.toString().toLowerCase()) ? Color.RED_500 : Color.BLACK}>
      {item.label}
    </Text>
  </Layout.Horizontal>
)
