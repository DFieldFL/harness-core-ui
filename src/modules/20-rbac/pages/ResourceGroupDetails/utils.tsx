import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import type { ResourceSelector } from 'services/cd-ng'

export const getSelectedResourcesMap = (
  resourceSelectorList?: ResourceSelector[]
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()

  resourceSelectorList?.map(resourceSelector => {
    if (resourceSelector.type === RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
      map.set(resourceSelector.resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    else map.set(resourceSelector.resourceType, resourceSelector.identifiers)
  })

  return map
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelector[] => {
  const resourceSelectors: ResourceSelector[] = []
  map.forEach((value, key) => {
    if (value === RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
      resourceSelectors.push({
        type: RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR,
        resourceType: key
      })
    else {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.STATIC_RESOURCE_SELECTOR,
        resourceType: key,
        identifiers: value
      })
    }
  })
  return resourceSelectors
}
