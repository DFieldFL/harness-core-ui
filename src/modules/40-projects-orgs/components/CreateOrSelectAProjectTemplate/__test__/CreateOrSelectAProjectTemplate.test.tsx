import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import { CreateOrSelectAProjectTemplate } from '../CreateOrSelectAProjectTemplate'

const props = {
  moduleDescription: 'CD Create or select a project',
  onSelectProject: jest.fn(),
  onCreateProject: jest.fn()
}
describe('Rendering', () => {
  test('should render', () => {
    const { container } = render(
      <TestWrapper>
        <CreateOrSelectAProjectTemplate {...props} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
