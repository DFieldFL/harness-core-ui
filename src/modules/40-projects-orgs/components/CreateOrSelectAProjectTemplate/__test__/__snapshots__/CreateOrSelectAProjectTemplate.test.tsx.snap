// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should render 1`] = `
<div>
  <div
    class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
  >
    <p
      class="StyledProps--font StyledProps--main"
      style="color: black; font-size: large; font-weight: bold;"
    >
      projectsOrgs.createOrSelectAProject
    </p>
    <p
      class="StyledProps--font StyledProps--main"
      style="color: black; font-size: small;"
    >
      projectsOrgs.getStarted
    </p>
    <div
      class="Layout--vertical Layout--layout-spacing-small StyledProps--main StyledProps--padding-top-large"
    >
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
        style="width: 60%;"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          projectsOrgs.selectAnExistingProject
        </span>
      </button>
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color"
        style="width: 60%;"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          projectsOrgs.createANewProject
        </span>
      </button>
    </div>
  </div>
</div>
`;
